app/models/course_template.rb

```diff
@@ -0,0 +1,15 @@
+class CourseTemplate < ActiveRecord::Base
+  validates :name,
+            presence: true,
+            uniqueness: true,
+            length: { within: 1..40 },
```

Comment:
Upper bound is same as already used in old Course model. I think these two should be same, as they represens same kind of object (and later courses can be created from these templates)